components/raftstore/src/store/snap.rs

```diff
@@ -207,7 +207,9 @@ fn retry_delete_snapshot(mgr: &SnapManagerCore, key: &SnapKey, snap: &Snapshot)
     false
 }
 
-fn gen_snapshot_meta(cf_files: &[CfFile], for_balance: bool) -> RaftStoreResult<SnapshotMeta> {
+// Create a SnapshotMeta that can be later put into RaftSnapshotData or written
+// into file.
+pub fn gen_snapshot_meta(cf_files: &[CfFile], for_balance: bool) -> RaftStoreResult<SnapshotMeta> {
```

Comment:
Is that still necessary?

Reply:
We need that to get a SnapshotMeta, which will be used by SnapshotMetadata in the "fake" MsgSnapshot

Reply:
This is for fast add peer, in which we compose am empty Snapshot to init a peer.